lib/normalize-query-options/geometry-filter.js

```diff
@@ -0,0 +1,46 @@
+const _ = require('lodash')
+const { normalizeArray } = require('./helpers')
+const transformEsriGeometry = require('../geometry/transfrom-esri-geometry-to-geojson-geometry')
+const transformCoordinateArrayToPolygon = require('../geometry/transform-coordinate-array-to-polygon')
+const transformEnvelopeToPolygon = require('../geometry/transform-envelope-to-polygon')
+const projectCoordinates = require('../geometry/project-coordinates')
+const normalizeGeometryFilterSpatialReference = require('./geometry-filter-spatial-reference')
+const normalizeSourceSR = require('./source-data-spatial-reference')
+
+function normalizeGeometryFilter (options = {}) {
+  const geometry = options.geometry || options.bbox
+
+  if (!geometry) return
+
+  const geometryFilterSpatialReference = normalizeGeometryFilterSpatialReference(options)
+
+  const geometryFilter = transformGeometryToPolygon(geometry)
+
+  const dataSpatialReference = normalizeSourceSR(options.sourceSR)
+
+  if (geometryFilterSpatialReference !== dataSpatialReference) {
+    geometryFilter.coordinates = projectCoordinates(geometryFilter.coordinates, {
+      fromSR: geometryFilterSpatialReference,
+      toSR: dataSpatialReference
+    })
+  }
+  return geometryFilter
+}
+
+function transformGeometryToPolygon (geometry) {
+  if (_.isString(geometry) || Array.isArray(geometry)) {
+    const coordinates = normalizeArray(geometry)
+    return transformCoordinateArrayToPolygon(coordinates)
+  }
+
+  if (geometry.xmin || geometry.xmin === 0) {
```

Comment:
why does it check `xmin === 0`?

Reply:
Also, you can use joi to validate the object. Maybe we can develop a joi library to contain all geometry object validations in the future, since we have seen so many...

Reply:
`xmin === 0` was in the old code, and I believe it is there to avoid a falsy check.  Without it, an `xmin` with a legitimate coordinate value of `0` would evaluate to `false` with only `if (geometry.xmin)`.

Using Joi is possible, but the tough part is validating something that can be so variable.  Perhaps just add a TODO?